ledger/participant-state/kvutils/tools/integrity-check/src/test/scala/ledger/participant/state/kvutils/tools/integritycheck/LogAppendingReadServiceFactorySpec.scala

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.tools.integritycheck
+
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import akka.stream.scaladsl.Sink
+import com.codahale.metrics.MetricRegistry
+import com.daml.ledger.validator.LedgerStateOperations
+import com.daml.metrics.Metrics
+import com.google.protobuf.ByteString
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{AsyncFunSuite, AsyncWordSpec, Matchers}
+
+@RunWith(classOf[JUnitRunner])
+class LogAppendingReadServiceFactorySpec extends AsyncWordSpec with Matchers {
+  "LogAppendingReadServiceFactory" should {
+    "handle empty blocks" in {
```

Comment:
It would be great to have a test case for the happy path as well.

Reply:
Done and good that you suggested this so I could check, because the test was not enabled and it was also broken...

Reply:
I should've noticed the @RunWith(...) annotation which is not used in this project.